From d15acfa3af4ab31ec8a80129514265b56897fd0d Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 11 Sep 2016 12:29:16 +0100
Subject: [PATCH 3/3] applets/icon-tasklist: Fix pinned icon squishing on login

As we override the behaviour of the base IconButton, we never queue_resize
with the initially windowless pinned icon buttons, meaning we're totally
squished and not obeying the 1:1.1 ratio.

This resolves issue #572

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 panel/applets/icon-tasklist/Buttons.vala | 1 +
 1 file changed, 1 insertion(+)

diff --git a/panel/applets/icon-tasklist/Buttons.vala b/panel/applets/icon-tasklist/Buttons.vala
index ef0cdfb..7bb1a0b 100644
--- a/panel/applets/icon-tasklist/Buttons.vala
+++ b/panel/applets/icon-tasklist/Buttons.vala
@@ -498,6 +498,7 @@ public class PinnedIconButton : IconButton
             return;
         }
         image.pixel_size = icon_size;
+        queue_resize();
     }
 
     public void reset()
-- 
2.10.0

